### PR TITLE
Integrate a section on auto-applied plugin

### PR DESCRIPTION
**Pull request by bmuschko**
The Wrapper version needs to modified to GA once it has been released. Needs to happen before merging the PR.

### CODE REVIEW

File: contents/index.adoc [hunk 1]
```diff
@@ -19,6 +19,18 @@ This guide shows you how to modify a build script to enable build scans. Optiona
 
 Gradle makes available a simple Java project that you can use to demonstrate build scan capabilities. If you wish to use it, clone or download the repository located at https://github.com/gradle/gradle-build-scan-quickstart. If you prefer to use your own build, you can skip this step.
 
+== Auto-apply the build-scan plugin
+
+Starting with Gradle 4.3 you can generate a build scan without any additional configuration in your build script. By using the command line option `--scan`, the latest build scan plugin is applied automatically. The build will ask you to accept the license agreement on the command line. The following console output demonstrates the behavior.
```
Comment (etiennestuder) [inline contents/index.adoc]: > By using the command line option `--scan`, the latest build scan plugin is applied automatically.

I would say `When using the command line option `--scan` to publish a build scan, the required `build scan` plugin is applied automatically. Before the end of the build, you are asked to accept...`.

File: contents/index.adoc [hunk 2]
```diff
@@ -19,6 +19,18 @@ This guide shows you how to modify a build script to enable build scans. Optiona
...
+
+----
+$ ./gradlew build --scan
+
+include::{samplesoutputdir}/auto-applied-build-scan-plugin/autoAppliedBuildScanPlugin.out[]
+----
+
+This mechanism makes it very easy to generate ad-hoc, one-off build scans without having to configure the plugin in your build. If you need more fine-grained configuration, then you can configure the build-scan plugin in a build or init script as described in the following sections.
```
Comment (etiennestuder) [inline contents/index.adoc]: `...without having to configure the build scan plugin in your build.`

Comment (etiennestuder) [inline contents/index.adoc]: `If you need finer grained configuration, you can...`(to be consistent with our wording in GE)

File: contents/index.adoc (hunk 1 above)
Comment (bmuschko) [reply contents/index.adoc]: done

File: contents/index.adoc (hunk 2 above)
Comment (bmuschko) [reply contents/index.adoc]: done

Comment (bmuschko) [reply contents/index.adoc]: done

**Comment by bmuschko** [comment]
@etiennestuder Thanks for your feedback. I addressed your comments. Please have another look.

**Comment by bmuschko** [comment]
@etiennestuder I will address your other comments in a separate PR and merge this one.

**Pull request merged**